### PR TITLE
PLAT 1669 : ABACShip

### PR DESCRIPTION
**Pull request by obidenko**
Signed-off-by: obidenko <obidenko@virtru.com>

[https://virtru.atlassian.net/browse/PLAT-1669](url)
Webpack and TS ready

### CODE REVIEW

File: examples/abacship-app/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+# User1
+
+U: `user1`
+P: `password`
+
+# User2
+
+U: `user2`
+P: `password`
```
Comment (elizabethhealy) [inline examples/abacship-app/README.md:25]: can you put the instructions to run it in the readme as well?

**Review by elizabethhealy — DISMISSED** [review]

File: examples/abacship-app/src/index.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+.revealed {
+    transform: rotateY(360deg);
+
+}
+
+.actual-value-hidden,.unknown-value-hidden {
+    opacity: 0;
+    display: none!important;
```
Comment (ivanovSPvirtru) [inline examples/abacship-app/src/index.scss:13]: Better not to use `important` but more specific selector. Like its visible if '.unknown-value' but invisible with '.unknown-value.unknown-value-hidden'

**Review by ivanovSPvirtru — APPROVED** [review]

File: examples/abacship-app/src/containers/GameDesk/utils.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+export function revealCell(rowIdx: number, colIdx: number) {
+  // actual hit
+  const data = getTDFData(rowIdx, colIdx);
+
+  if (data === "ocean") return CELL_TYPE.OCEAN;
+  if (data === "aircraft carrier" || data === "battleship" ||
+  data === "cruiser" || data === "destroyer" ||
+  data === "submarine") return CELL_TYPE.PLAYER_ONE;
+  if (data === "aircraft carrier" || data === "battleship" ||
```
Comment (dmihalcik-virtru) [inline examples/abacship-app/src/containers/GameDesk/utils.ts:26]: ???

File: examples/abacship-app/src/containers/GameDesk/GameDesk.tsx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+    if (currentServerStatus === ServerStatus.p2_request_attr_from_p1 && playerData.name === "player1") {
+      //PLAYER 2
+      shareAccess();
+    }
+
+    // PLAYER TURN
+    if (currentServerStatus === ServerStatus.p1_turn) {
+      // PLAYER 1
+
```
Comment (dmihalcik-virtru) [inline examples/abacship-app/src/containers/GameDesk/GameDesk.tsx:109]: Are there some TODO items missing here?

**Review by dmihalcik-virtru — APPROVED** [review]

**Comment by cassandrabailey293** [comment]
tested and it works! 🎉 
@elizabethhealy will make a few changes - changing the ABACSHIP font, removing the virtru and telsa logos from image assets, and adding a few things to the troubleshooting section of the readme. Then we're good to merge this

**Review by cassandrabailey293 — APPROVED** [review]

**Pull request merged**